modules/calib3d/src/solvepnp.cpp

```diff
@@ -311,19 +311,23 @@ bool solvePnPRansac(InputArray _opoints, InputArray _ipoints,
 
     opoints_inliers.resize(npoints1);
     ipoints_inliers.resize(npoints1);
-    result = solvePnP(opoints_inliers, ipoints_inliers, cameraMatrix,
-                      distCoeffs, rvec, tvec, useExtrinsicGuess,
-                      (flags == SOLVEPNP_P3P || flags == SOLVEPNP_AP3P) ? SOLVEPNP_EPNP : flags) ? 1 : -1;
+    try
+    {
+        result = solvePnP(opoints_inliers, ipoints_inliers, cameraMatrix,
+                          distCoeffs, rvec, tvec, useExtrinsicGuess,
+                          (flags == SOLVEPNP_P3P || flags == SOLVEPNP_AP3P) ? SOLVEPNP_EPNP : flags) ? 1 : -1;
+    }
+    catch (const cv::Exception& e)
+    {
+        result = 0;
+        CV_UNUSED(e);
```

Comment:
Is it needed since it is used in `CV_LOG_DEBUG`?

Reply:
I'm not quite sure about this, but I have seen same usage in [calibinit.cpp](https://github.com/opencv/opencv/blob/cac5b0fe5a3e06718a6cb408d6468ff87428ef2a/modules/calib3d/src/calibinit.cpp#L2261-L2266). Maybe in Release build, `CV_LOG_DEBUG` doesn't use `e`, then `CV_UNUSED` can supress the warning?

Reply:
Yes, I think that it is to probably suppress warning in release mode.